shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -152,23 +152,24 @@ public void createAndroidViewForPlatformView(
             throw new IllegalStateException(
                 "The Android view returned from PlatformView#getView() was already added to a parent view.");
           }
-          platformViews.put(request.viewId, view);
```

Comment:
nit: what about moving the checks `if (view == null)` and `if(view.getParent() != null)` to `initializePlatformViewIfNeeded` ?

Reply:
done